packages/flutter/test/widgets/widget_inspector_test.dart

```diff
@@ -846,6 +850,26 @@ class _TestWidgetInspectorService extends TestWidgetInspectorService {
       );
     });
 
+    testWidgets("WidgetInspector isn't inserted into the widget tree when "
+        'debugWillManuallyInjectWidgetInspector is set', (WidgetTester tester) async {
+      final TestWidgetsFlutterBinding binding = TestWidgetsFlutterBinding.ensureInitialized();
+      // Disable automatic insertion of WidgetInspector into the widget tree by WidgetsApp.
+      binding.debugWillManuallyInjectWidgetInspector = true;
+
+      await tester.pumpWidget(WidgetsApp(color: Colors.red, builder: (_, _) => const Text('Foo')));
+
+      // Verify that the widget inspector is disabled and there's no instances of WidgetInspector
+      // in the tree.
+      final Finder widgetInspectorFinder = find.byType(WidgetInspector);
+      expect(binding.debugShowWidgetInspectorOverride, false);
```

Comment:
Why can't we use the existing `debugShowWidgetInspectorOverride` for widget previews instead?

Reply:
We do, but we also need to make sure that `debugShowWidgetInspectorOverride == true` doesn't result in an instance of `WidgetInspector` being inserted by `WidgetsApp`.

This test is simply testing that `debugWillManuallyInjectWidgetInspector` is able to disable the creation of `WidgetInspector` by `WidgetsApp`.

Reply:
I'm trying to understand if we can set `debugShowWidgetInspectorOverride` to `false` to prevent the widget inspector from being injected into the `WidgetsApp` (how is the behavior of `debugShowWidgetInspectorOverride` different than `debugWillManuallyInjectWidgetInspector`)?

Reply:
We can, but the problem is we also want to allow for the manually created `WidgetInspector` widgets to be removed from the widget tree when widget inspection isn't enabled. That's done by listening to the value of `debugShowWidgetInspectorOverride`, which DevTools updates through a service extension to enable / disable the inspector.